### DISCUSSION TITLE
Inappropriate 'Return to Previous Style' tool tip

### DISCUSSION THREAD

**Discussion by artisteacher** [opentoonz/opentoonz#6543] [General]
If auto apply is active in the style editor, the color swatch for the previous style displays as the current style.
When auto apply is off, you can return to the previous style:
<img width="527" alt="screen shot 2017-08-08 at 3 31 04 pm" src="https://user-images.githubusercontent.com/21159116/29090867-ff06f9ce-7c4e-11e7-8bbb-a00d5dda8517.png">
When auto apply is on, the current style displays twice (really evident in the MyPaint brushes) and you can't return to the previous style through the swatch.
<img width="528" alt="screen shot 2017-08-08 at 3 31 45 pm" src="https://user-images.githubusercontent.com/21159116/29090948-469473a2-7c4f-11e7-9476-9254fc63b5b7.png">

**Comment by ghost** [comment]
I'm not sure this is a bug.  I was playing with the code for quite a bit tonight and when a color is applied, it becomes the current style and the previous style.  If the previous style stays, it can actually get confusing at a quick glance what color you are using.  It will almost definitely be unclear to new users why there are two colors shown.
![double](https://user-images.githubusercontent.com/4576381/29237573-5f3e0256-7ede-11e7-893a-019b18ecfd72.JPG)
It is possible to make the previous style stick around, but I'm not sure it is the best for users.

**Comment by artisteacher** [comment]
In that case, when auto-apply is on, maybe only one style should display.

**Comment by gab3d** [comment]
I agree with not showing previous style when Auto Apply is activated

El ago. 12, 2017 2:30, "artisteacher" <notifications@github.com> escribió:

In that case, when auto-apply is on, maybe only one style should display.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/opentoonz/opentoonz/issues/1387#issuecomment-321959023>,
or mute the thread

**Comment by RodneyBaker** [comment]
>
> It is possible to make the previous style stick around, but I'm not sure
> it is the best for users.


I do get the sense that the dual nature of the visible swatch is going away.

I hate to lose features and I am noting that through the course of
incremental improvement we are losing several useful features without
anyone realizing they didn't get the chance to say "goodbye".
The idea of being able to see the previous style is a good one although I
must acknowledge I didn't even know that feature existed until this round
of changes.

*Perhaps that portion of the previous style's preview could be made to be a
tiny slice (much more vertical than wide)?  Or perhaps the current display
with a slightly diagonal frame/line (subtly suggesting movement/time).*

*Another option would be to place some text there that states "previous"
and "current" although I'm not sure that'd be optimal.  At least the intent
would be clearer.*

Keep in mind that much of this we are considering as confusing relates to
workflow with Auto Apply turned off.
I'm not sure of the steps of that workflow which would make such a thing
optimal but for someone whose job is solely to create a productions
required styles so that others can utilize... that display of previous
style might be quite important.

It has been so long since I've used the Style Editor without Auto Apply on
that I probably should try that.

**Comment by artisteacher** [comment]
If it's not possible to switch to a single swatch when auto-apply is on, it would be helpful to only include the "return to previous style" tool tip when auto-apply is off.

**Comment by RodneyBaker** [comment]
The ability to return to previous style is a useful feature that I had forgotten about because I only use auto-apply.

I'm retagging this as 'Feature Request' largely because there currently is no logic in the code for separating the displayed styles from when Auto-Apply is on versus when it is off.

Here however for the record is the location where applicable code is found. 
https://github.com/opentoonz/opentoonz/blob/aecc26c18e2fe4debcada5cdaedca3cc11c04ef9/toonz/sources/toonzqt/styleeditor.cpp#L3102

Aside:  The logic isn't there for this either but it would be useful if the 'Previous/Current' style displayed was aware of color style keyframes.   Of course then users would want three displays... one for previous, one for current and one for next available style (via style keyframes that is).

**Comment by RodneyBaker** [comment]
Updated labeling.
Now that feature requests are accepted in OT github this fits more into bug category than as a feature request.
